### ISSUE TITLE
What does a WPub URL resolve to?

### ISSUE THREAD

**Issue by prototypo** [w3c/wpub#94]
# Problem Statement

We have decided that a WPub with be identified with a URL. What should that URL resolve to?

This issue suggests four scenarios and evaluates their ramifications in the hopes of driving toward consensus. A conclusion section at the bottom of this post suggests an answer to the question posed.

# Scenarios

The working group has considered several possible answers to this question. They mostly fall into one of four scenarios:

1. A WPub URL resolves to a JSON manifest file
2. A WPub URL resolves to an HTML file that contains both a table of contents (TOC) and other metadata
3. A WPub URL resolves to an HTML file containing a TOC and which links to a JSON manifest file
4. A WPub URL resolves to a binary package (e.g. a ZIP file or SQLite database file)

Each of these scenarios are considered in turn as they would be viewed by four types of clients:

- An existing search engine bot
- A WPub aware search engine bot
- A reader using an existing user agent
- A reader using a WPub aware user agent

## Scenario 1: JSON Manifest

If a WPub URL resolves to a JSON manifest file, the four clients may be expected to act like this:

- An existing search engine bot (path 1A in the image below) will not index JSON files.
- A WPub aware search engine bot (path 1B in the image below) will need to index all JSON files on the Web to determine which ones represent WPubs.
- A reader using an existing user agent (path 1C in the image below) will be presented with a raw JSON representation.
- A reader using a WPub aware user agent (path 1D in the image below) will be able to see and operate upon the WPub as intended.

Scenario 1 seems inconvenient for both old and new clients with the exception of new WPub aware user agents.

![wpub-canonicalresourcejson](https://user-images.githubusercontent.com/824236/32531863-3599fdee-c3fb-11e7-8d93-e2261c34278c.png)

## Scenario 2: HTML TOC & Metadata

If a WPub URL resolves to an HTML file that contains both a table of contents (TOC) and other metadata, the four clients may be expected to act like this:

- An existing search engine bot (path 2A in the image below) will encounter and index the HTML page without modification.
- A WPub aware search engine bot (path 2B in the image below) will be able to determine that the HTML page represents a WPub, but will need to parse the entire file in a WPub-specified manner to extract metadata.
- A reader using an existing user agent (path 2C in the image below) will see a TOC containing links to the components of the WPub.
- A reader using a WPub aware user agent (path 2D in the image below) will be able to see and operate upon the WPub as intended.

![wpub-canonicalresourcehtml](https://user-images.githubusercontent.com/824236/32531901-774e1a5e-c3fb-11e7-92b9-420a266c7a5d.png)

Scenario 2 would be handled cleanly for old and new clients, but could cause difficulties when old clients are presented with metadata they cannot understand. There is also some danger of overloading or stretching the use of HTML to define the necessary metadata within an HTML document. New clients would need to parse the metadata out of the HTML to operate upon it.

## Scenario 3: HTML TOC & JSON Manifest

If a WPub URL resolves to an HTML file containing a TOC and which links to a JSON manifest file, the four clients may be expected to act like this:

- An existing search engine bot (path 3A in the image below) will encounter and index the HTML page without modification.
- A WPub aware search engine bot (path 3B in the image below) will be able to determine that the HTML page represents a WPub, and follow the provided link to the metadata if it wishes.
- A reader using an existing user agent (path 3C in the image below) will see a TOC containing links to the components of the WPub.
- A reader using a WPub aware user agent (path 3D in the image below) will be able to see and operate upon the WPub as intended.

![wpub-canonicalresourcehtmljson](https://user-images.githubusercontent.com/824236/32531914-8e03df40-c3fb-11e7-8ab0-59319829a0d0.png)

Scenario 3 seems to cleanly handle old and new clients in appropriate ways. Old clients could follow their noses to the components of a WPub, and new clients could easily load the JSON object to efficiently access metadata.

## Scenario 4: Binary WPub

If a WPub URL resolves to a binary file, the four clients may be expected to act like this:

- An existing search engine bot (path 4A in the image below) will either be aware of the MIME type for the binary file and attempt to handle it in an inappropriate manner, or it will need to be modified to handle a new MIME type.
- A WPub aware search engine bot (path 4B in the image below) will either be confused by the overloading of an existing MIME type, or handle a new MIME type correctly.
- A reader using an existing user agent (path 4C in the image below) will probably be asked to download the binary.
- A reader using a WPub aware user agent (path 4D in the image below) will either be confused by the overloading of an existing MIME type, or handle a new MIME type correctly.

Scenario 4 could lead to confusion for both old and new clients unless a new MIME type is registered.

![wpub-canonicalresourcebinary](https://user-images.githubusercontent.com/824236/32531923-9e4066f8-c3fb-11e7-8704-7c8c214c14c2.png)

# My Conclusion

Given the various pros and cons, **Scenario 3** (the URL to a WPub resolves to an HTML TOC document, which in turn links to a JSON manifest) seems to most easily interoperate with the existing Web and provides a clean upgrade path to WPub-aware clients.

# Acknowledgements

Thanks to @BigBlueHat @TzviyaSiegman @iherman @tcole3 @bdugas @GarthConboy and others from the Publishing WG for discussions which led to this issue and its discussion.

**Comment by mattgarrish** [comment]
Do we need to require toc in this, or do we only need to require that the link resolve to a resource that is considered the primary entry point for the publication, and that must include a link to the manifest?

It sounds like we're mandating the presence of a table of contents, when not every publication will need one. Using the table of contents, or ensuring the landing document has a clear link to one, is perhaps only best practice in the case of multi-document publications?

**Comment by dauwhe** [comment]
Yeah, this is one of my fundamental questions. If I point my browser at the URL of a WP, what happens?

```
> GET /MobyDick/ HTTP/1.1
> Host: www.example.com
> User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/51.0.2704.106 Safari/537.36 OPR/38.0.2220.41
> Accept: */*
```

Strongly agree that the URL must resolve to an HTML document, and that this HTML document must contain a link to the manifest (or contain the manifest; I think that's still an open question). 

Like @mattgarrish, I'm a little less certain about requiring this document to contain a TOC. Perhaps if it doesn't contain a TOC, and it is a multiple-document publication, it must link to a TOC? Sadly `rel=contents` isn't super-standard, but `rel=index` might be appropriate? Or is it enough that the manifest include a link to a TOC?

Do we require that this be the first document in the default reading order? I think yes. In retrospect, the whole "begin reading" thing in EPUB felt like an evasion. Saying that [1] front matter is so important that it must come at the front of the book, and [2] that it is so unimportant that we don't want the reader to actually see it, is just avoiding responsibility for designing your own content appropriately.

**Comment by iherman** [comment]
To repeat some arguments, just for the records, against option 3:

- If the metadata used the HTML elements like `<meta>`, `<link>`, or even `<title>`, that would be, in my view, in violation to the HTML spec. That spec clearly says "The `meta` element can represent document-level metadata". I.e., using that element for the Publication, instead of the for the containing document would be misusing those for a different purpose. We should not do that.
- It would be possible, to include the whole metadata in a `<script>` with the type set for javascript but, in practice, this may be a stretch. For example, no HTML authoring tool that I know would make that easy to do. Also, the JSON metadata should then me made in a way (using JSON-LD tricks) that the subject for the metadata in question is the Publication and the the enclosing document itself.

My conclusion on option 3 that it should _not_ be adopted.

**Comment by dauwhe** [comment]
> If the metadata used the HTML elements like <meta>, <link>, or even <title>, that would be, in my view, in violation to the HTML spec. That spec clearly says "The meta element can represent document-level metadata". I.e., using that element for the Publication, instead of the for the containing document would be misusing those for a different purpose. We should not do that.

I don't think option 3 says anything about publication metadata being expressed in the HTML document that the WPUB URL resolves to. Couldn't that metadata be in the manifest, or linked to from the manifest?

**Comment by iherman** [comment]
Oh bugger, I wanted to say Option 2! I will edit the comment...

(Lesson: never say anything serious before breakfast!)

**Comment by rdeltour** [comment]
> If the metadata used the HTML elements like <meta>, <link>, or even <title>, that would be, in my view, in violation to the HTML spec. That spec clearly says "The meta element can represent document-level metadata". I.e., using that element for the Publication, instead of the for the containing document would be misusing those for a different purpose. We should not do that.

There's a precedent however in web apps, where HTML’s `meta` elements are very frequently used to represent app-level metadata.
So I don't think the "violation of the HTML spec" applies is very significant. Said spec can be updated to pave the web apps cowpaths.

**Comment by iherman** [comment]
> There's a precedent however in web apps, where HTML’s meta elements are very frequently used to represent app-level metadata.
> So I don't think the "violation of the HTML spec" applies is very significant. Said spec can be updated to pave the web apps cowpaths.

Maybe there are such apps in the wild, and I do not think what they do is correct. Maybe, but only maybe, the Web Platform WG will, at some point in the distant future, pave that particular cowpath (I cannot judge how wide this usage is), but I do not think this WG should adopt an approach which is in violation with the specifications as of today.

**Comment by iherman** [comment]
(Just to make it clear: my personal vote goes firmly for option 3...)

**Comment by mattgarrish** [comment]
> I do not think this WG should adopt an approach which is in violation with the specifications as of today

I'm kind of torn on this, as we're already doing similar in allowing certain information to be harvested from the content (title, language, etc.). It's (arguably) useful information for legacy browsers that don't support the publication.

But I agree to the extent that we should not be confusing people that it's either/or. Maintaining metadata in multiple places is always a disaster in waiting. WAM gives precedence to manifest metadata and we should do the same.

I prefer Ivan's approach, but can live with what we've done. At a minimum, we can never stop people from expressing whatever metadata they want wherever they want.

**Comment by WSchindler** [comment]
Considering the 4 scenarios I think option 3 seems the best solution to accommodate all types of clients. From a user perspective, we would always need some kind of landing page for the WP. Typically, it should contain a **TOC** (via the `<nav>` element) to enable the user to visually access the different parts of the WP - i.e. those that are part of the reading order - either sequentially or in a random order. If it doesn't contain a TOC and has more than one constituent documents, we would still need some starting point. 
Would you agree that any resource listed in the reading order should have a link to the one and only manifest of a WP which contains all the information for a WP-aware user agent to properly consume and render a WP?

**Comment by dauwhe** [comment]
> To say 'the landing page is the first document in the reading order may not be good either: the content may not be in HTML. We had this discussion on the TPAC F2F of publications consisting of drawings or audio files only. For those cases, the TOC is helpful for non WPUB aware browsers.

Ah, this helps clarify that I have a concern here. If this HTML document is not the first document in the default reading order, then a WP-aware user agent will present an entirely different resource to the user than a non-WP-aware user agent. Quick example: if my WP has this (very common) order:

```
cover.html (embedded in html)
title-page.html
toc.html
chapter-1.html
chapter-2.html
```

I might set things up so that the URL for the WP resolves to `cover.html` and that contains the link to the manifest. But what if the manifest says that the first item of the default reading order is `chapter-1.html`? Then a non-aware UA would present the cover, and a WP-aware UA would present ch1? I find that very confusing. 

I agree that having the URL resolve to a TOC is ideal. You can always hide it visually. 

I would also argue that something in the WP has to be HTML (or something that supports links). I don't see how you make a JSON manifest + audio/image files only work in option 3, because otherwise a non-aware UA cannot access the content through the WP URL.

**Comment by mattgarrish** [comment]
> the landing page contains the TOC

But why enforce this? What does it accomplish, really?

What if I want my landing page to be the cover page and clicking on the cover page takes you to the table of contents?

**Comment by rdeltour** [comment]
> Maybe there are such apps in the wild, and I do not think what they do is correct. (...) I do not think this WG should adopt an approach which is in violation with the specifications as of today.

I'm only saying I don't think that using `meta` would necessarily be a violation. I mean, `application-name` is part of the standard metadata names _defined in the HTML spec_, and can hardly be taken for document-level metadata.
It's a blurry line, you can always argue that a piece of metadata applies to the document when it gives info on the larger object (app, publication) this document belongs to.

That said, my vote goes to option 3. I'm just thinking that it doesn't necessarily precludes mixing it with option 2 to widen UA support, like web apps do: have an authoritative external manifest, but also fallback to HTML metadata when relevant.

**Comment by dauwhe** [comment]
> Why would the UA change the document you've navigated to? If the reading order forces you away from the resource you want to view, that's a very bad thing.

I suppose that's what I'm wondering, but perhaps this is more a lifecycle & implementation question. A WP-aware UA opens the HTML resource at the WP URL, and processes the manifest. What is it obligated to do then? I suppose it could present a "begin reading" button that would cause a navigation. Maybe it just creates a navigation/personalization overlay (as Readium Cloud Viewer does). 

Maybe my point is that it would be silly to have the URL point to something other than what you want the reader to see first. Even RFC 6919 doesn't have formal language for that :)

**Comment by mattgarrish** [comment]
> A WP-aware UA opens the HTML resource at the WP URL, and processes the manifest. What is it obligated to do then?

Right, this is where we (probably) won't find consensus. I'm of the mind that the user agent shouldn't change the resource you're on when it initiates the reading experience. But if it does, I also don't have any problem with that as user's will decide the fate of such a feature. I just don't think such a change should happen without prompting, in which case both the vanilla and enhanced experiences are ultimately the same, there's just an extra opt-in from the user that changes one scenario.

I think it would also help to make clear that we're trying to accommodate many reading scenarios somewhere (browser, app (in browser), polyfilled publication, publication as app), but let's not discuss that in this thread.

**Comment by iherman** [comment]
>> the landing page contains the TOC
>
> But why enforce this? What does it accomplish, really?

What I think we should avoid is that the same information item could be expressed in different places. I am afraid of the possible confusion that would result from that (what if they are defined in different places? Which one has priority? What if the information are conflicting? etc.)

To be more specific: we have a set of information items that must be serialized. My mental model is:

1. One, and only one, of the information item appears in the landing page, and _the landing page only_
2. _All other_ information items are serialized in JSON (as we decided) and are, therefore, part of the manifest (waybill?) file that is linked from the landing page.

In other words, there is no ambiguity on where a specific information item is defined.

_If_ this model works that we must choose which information item is covered by (1). The TOC seems to be the best choice..

**Comment by mattgarrish** [comment]
> One, and only one, of the information item appears in the landing page, and the landing page only

I agree and disagree. There is one piece of information that must appear on the landing page: the link to the manifest. It can appear elsewhere, but whether it has to we keep changing course on (I preferred our original recommendation, so don't object to the subsequent PR that's appeared.)

My fears may be overblown, granted, but I think when we stray into mandating content we're going down a potentially restrictive and unwanted path. I'm not going to lie down in the road over this, though.

**Comment by dauwhe** [comment]
So do we have rough consensus that:

1. The URL of a WP must resolve to a resource (the "landing page") which
2. must contains a link to the manifest and
3. should contain a `<nav>`?

**Comment by bduga** [comment]
I am not sure we need to express metadata level information as meta elements, as I agree with Ivan that seem like appropriating file level metadata. But there is also the option of simply encoding all the metadata in HTML, by (ab)using classes (or another mechanism). That gives us full markup in the metadata, as well as CSS styling. And it means the content can be suppressed for display via CSS. So the landing page could have the text of the title, list of authors, etc that are also displayed directly to the user without duplication (there isn't an author in the metadata and an author on the landing page).

**Comment by prototypo** [comment]
Oops, I meant to acknowledge @bduga - apologies!

**Comment by azaroth42** [comment]
From the cheap seats at the back of #W3CTPAC ... (over several sessions) ...

I agree with @mattgarrish. In my experience with the development of the [IIIF manifest](http://iiif.io/api/presentation/2.1/) spec, the link to the JSON is of course the most important. Other useful information is the position to enter the document's structure, if not the beginning, allowing multiple entry points with different behavior.

**Comment by GarthConboy** [comment]
I'm (also) okay with options #2 and #3.  Per Dave's "should contain a <nav>?" above, if a "<nav>" is a MUST somewhere, it seems it's better to be a MUST in a particular place such that we don't introduce opportunities for it two be in multiple places and having the various instances fight.

**Comment by mattgarrish** [comment]
What we have in the infoset is that you can specify the toc as a property or you can reference the html nav element that contains it. I think that's optimal, as it doesn't require the toc to be in any particular place, and doesn't necessitate duplication. It also allows different tocs for content and rs if you so desire.

**Comment by iherman** [comment]
At this point I believe we indeed have a consensus as described in [Dave's comment](https://github.com/w3c/wpub/issues/94#issuecomment-342903893), and we do not fully agree what the landing page would really contain. I would propose, just for the sake of moving ahead, to put that into the document (in the form of a PR) and open a separate issue on, specifically, "MUST the TOC appear on the landing page" which we will have to cover at some point.

**Comment by iherman** [comment]
Added a PR (#98). I have not opened a new issue; should be done only if the PR is accepted for merge.

**Comment by prototypo** [comment]
Should this issue be closed now that PR [#98](https://github.com/w3c/wpub/pull/98) has been merged?

**Comment by iherman** [comment]
#98 has not been merged yet...:-) 

Yes, this should be closed when it is merged.

**Comment by HadrienGardeur** [comment]
Frankly, I don't think that we should enforce anything.

If the URL is primarily meant to identify, it doesn't really matter if it points to a manifest, a TOC or a resource of the publication.

All these resources will have their own URL anyway, and since we can't expect all publications to have a TOC, I really don't like the proposed resolution.

**Comment by llemeurfr** [comment]
EDRLab's position is that the WPub URL should resolve to a **"landing page"**, ie. an HTML page that introduces the publication to the reader.  It MUST contain in its header a link to the JSON Manifest (which has its own URL). It MAY contain an HTML TOC, but it is not mandatory. Such a TOC MAY be machine readible if its structure follows some best practices yes to be defined, but it is not mandatory. It may also present other information, like those found on any bookselling website: a title, authors, cover, description, extracts of reviews etc. Such content is not defined by the spec, it may or may not be machine readable (it may if eg. but WPub author integrates Rich Snippets in the HTML structure, but the WPub spec is silent on this). 

Therefore:
- a search engine, will be able to index this landing page (if the landing page is freely accessible).
- a user will be able to navigate through the WPub from this landing page if it contains links to some parts of the content (it may be a full TOC, or simply a link to the preface).
- a WPub aware user agent will follow the link to the WPub manifest and act accordingly. 

In a package, this landing page will be included so that a round-trip btw PW and PWP can be facilitated. But the JSON manifest will have a conventional name, therefore the PWP user agent will be able to jump to the manifest without having to go through the landing page.

**Comment by iherman** [comment]
@llemeurfr, I agree, with the exception of 'should': WPub URL MUST resolve to a landing page. If we want WP-s to be handled by current browsers and not go completely blind on the identifier of a WPUB (or, worse, displaying an unintelligible JSON file), we have to have this. 

Architecturally it would be nicer if we did not have this requirement. But practicality is what it is...

**Comment by HadrienGardeur** [comment]
Why do we assume that the WPub URL will be the primary way of sharing and discovering a WPub ? This is a very different requirement than identifying a publication.

Also, assuming that one URL is more important than any other is IMO the wrong way to think about this.
We'll have multiple URLs that have a specific role instead, each identified by a rel value:

- `manifest` for the JSON manifest
- `start` for the beginning of the publication
- `index` or `contents` for the TOC

`identifier` is also a candidate for the IETF link registry: https://tools.ietf.org/html/draft-vandesompel-identifier-00

By mixing up two different requirements we're not really making things clear at all.

**Comment by HadrienGardeur** [comment]
I haven't read a strong case so far to justify that the WPub URL (whatever that is) should be any of the above.

Many comments in this thread talk about a "landing page".  It's very unclear to me why we jumped to the point where WPub URL = landing page, but even if we assume that's the consensus, there are many questions worth asking about such a page.

Why should a landing page be the same thing as the table of contents? Why should a landing page contain a link to the manifest?

As a publisher, if my publication is behind a paywall, isn't it better to have a landing page that shows basic info about the book plus a giant buy button instead of being forced to link to a manifest + include a table of contents (which might not even exist)?
What if my publication only exists as a PWP, how do I link to the manifest in the first place if I don't have one?

**Comment by iherman** [comment]
@HadrienGardeur 

> WPub URL will be the primary way of sharing and discovering a WPub

We are on the Web. That means URLs _are_ the primary way of sharing and discovering things. Various links with rel values are nice and maybe necessary in some circumstances, but on the Web a simple (ie, "unqualified") _is_ the way to discover things indeed. 

> Why should a landing page contain a link to the manifest? Do we need the landing page to be part of the publication at all?

See the [problem statement](https://github.com/w3c/wpub/issues/94#issue-272078084) of @prototypo: how else would the discovery of a Web Publication happen based on its URL? Put it another way, what should an HTTP request of the WPUB's URL return? @prototypo has laid out the various alternatives…

> Why should a landing page be the same thing as the table of contents?

It is not the same thing. A landing page may include other information, header, whatever (see, eg, [a previous comment](https://github.com/w3c/wpub/issues/94#issuecomment-343955531)). The only thing that is necessary is the link to the manifest. This is how a WPUB-aware browser, or a browser with an appropriate javascript running running on top, would find out the WPUB itself. Adding a TOC is only a convenience, although sounds like a useful convenience to me. A landing page may be without it.

> As a publisher, if my publication is behind a paywall, isn't it better to have a landing page that shows basic info about the book plus a giant buy button instead of being forced to link to a manifest + include a table of contents (which might not even exist)?

As I said, it is possible show basic info and any types of button. But how would the browser find the manifest if not via a link?

> What if my publication only exists as a PWP, how do I link to the manifest in the first place if I don't have one? 

Although we indeed did not discuss the PWP-only case, I am not sure what you mean by a PWP without a manifest. To my mind, at this point, a WP as well as a PWP MUST have a manifest. That makes it a publication...

**Comment by llemeurfr** [comment]
I tried to find examples of what I think landing pages should look like. Think about a simplified version of:
https://www.goodreads.com/book/show/1318.Last_of_the_Amazons?from_search=true 
or 
https://www.babelio.com/livres/Zuber-Facebook-ma-tuer/247349 

Replace the calls to action by link to the "Table of Contents" (as @dauwhe said, sometime its better to have the TOC as a spine item) or a "Read the book" (which could link to the cover), keep only a few reviews, keep the links to other WPubs (recommandations), even the call to rate the book (why not?) and you've got the idea.

Such landing page should no be behind a paywall, and it must not be in the spine (i.e. reading order), as it is "outside" of the book.
As I said earlier, a reading system does not need this landing page to find the manifest in a PWP.
If a PWP is created in a standard publishing workflow, the PWP may contain NO landing page. 
In such a case, moving the PWP to the Web as a WPub will involve some work (either machine of human based).

**Comment by iherman** [comment]
@llemeurfr, I must admit that is not the way I view things. For me, the landing page is an integral part of a WP, and is independent of the page set up by, in this case, a distributor. It may contain information that the publisher want to convey, but it is a bit like the cover page of a usual book.

> If a PWP is created in a standard publishing workflow, the PWP may contain NO landing page.

I do not see it that way. Maybe the term "landing page" is not a good term for a PWP, but the way I see a PWP is that it _may_ be expanded on a Web site and, at that point, it is a perfectly valid WP without any further action.

I _think_ that we may have here a (further) possible difference between what we call a PWP and what we call EPUB4. Yes, maybe, an EPUB4 can be one step further away from a WP insofar as the manifest is at an agreed-upon place within the package (much like there are such files in EPUB3), and then the landing page concept may not be necessary. But, as you say, turning this into a WP would require more than just unpacking it on the Web site.

**Comment by baldurbjarnason** [comment]
We're already at a point where PWPs are going to be substantially different from regular WPs, even if just from a security perspective (you can't do away with HTTP-addressable locations without losing compatibility with the open web stack).

So, with that in mind I'm fine with WPs and PWPs (and EPUB4) having differences when it comes to landing pages. They're going to have a _lot_ of incompatibilities anyway.

I do worry a bit whether requiring the WP URL to be a landing page will land us in trouble in the future just on the basis that hard coupling is generally bad, but don't really have an argument against it that hasn't already been brought up in this discussion.

From the Pressbooks perspective, where I've been looking into the task of prototyping web-publications, we already have landing pages anyway for each book so this particular feature wouldn't cause undue implementation difficulties.

**Comment by HadrienGardeur** [comment]
> We are on the Web. That means URLs are the primary way of sharing and discovering things. Various links with rel values are nice and maybe necessary in some circumstances, but on the Web a simple (ie, "unqualified") is the way to discover things indeed.

Right, and on the Web any URL works. A website is not necessarily accessed from a single URL, there are many options available.

> @llemeurfr, I must admit that is not the way I view things. For me, the landing page is an integral part of a WP, and is independent of the page set up by, in this case, a distributor. It may contain information that the publisher want to convey, but it is a bit like the cover page of a usual book.

That part is still very unclear. Is the landing page part of the publication (a primary resource or whatever we call them these days) or not?

Expecting all PWP (and therefore all EPUB 4) to have a landing page on the Web is IMO completely unreasonable, and expecting such landing pages to have a TOC or a manifest linked in there is even more unreasonable.

**Comment by laudrain** [comment]
@HadrienGardeur: all books/ebooks have a Web page to describe them on publisher Web sites at list. We are not far from a landing page.

**Comment by HadrienGardeur** [comment]
@laudrain but we're not talking about books (this is about publications, a much broader category), and not all publishers have a landing page per book, far from it.

**Comment by laudrain** [comment]
@HadrienGardeur so they will have when they will adopt WP/PWP/EPUB4.

**Comment by HadrienGardeur** [comment]
@laudrain this puts a massive barrier to entry on top of EPUB4 compared to previous versions of EPUB, which IMO is a non-starter.

**Comment by iherman** [comment]
> this puts a massive barrier to entry in front of EPUB4 adoption compared to previous versions of EPUB, which IMO is a non-starter.

I think that is an overstatement. EPUB3 has a massive barrier in terms of information that has to be provided (including a nav file), this is less, not more...

**Comment by HadrienGardeur** [comment]
@iherman it's far from an overstatement and I agree that EPUB3 took a few wrong turns down the road btw.

The main issue here is that requiring a landing page is completely disconnected from current EPUB production. In many cases, the people producing an EPUB file would not be in the position to create and/or host such a landing page at all.

For a company like Hachette, sure that's something that they can deal with, but we're talking about making it easy for anyone to produce an EPUB4, not just the happy few.

**Comment by iherman** [comment]
@HadrienGardeur I honestly do not understand. The landing page can be as simple as an empty HTML page with a link to the manifest. Or, if the publication is a single document, it can be the document proper with a link to the manifest. The only thing we say is that the URL of a WP should lead to an HTML page that has a link to the manifest. Anything more is icing on the cake.

**Comment by HadrienGardeur** [comment]
@iherman If the publication only exists as a PWP/EPUB4, there won't be a URL for the manifest in the first place or any resource from the PWP that you can reference using a URL either.

**Comment by iherman** [comment]
> @iherman <https://github.com/iherman> If the publication only exists as a PWP/EPUB4, there won't be a URL for the manifest in the first place or any resource from the PWP that you can reference using a URL either.
> 


As I said, this may be a difference between EPUB4 and (P)WP.

My (informal and personal) definition of a PWP is that it is a package that, if I unpack onto the Web (even if it is on localhost and serve it in the browser through a server running on my machine), it is a valid WP. (The place where I unpack it providing a natural address, which may be different of the identifier, but that is a longer story). I am o.k. by relaxing this for EPUB4, which _may_ have a landing page, but again may not. 

At this moment, we are talking about a WP, though.

**Comment by llemeurfr** [comment]
>  I am o.k. by relaxing this for EPUB4, which _may_ have a landing page, but again may not.

@iherman, if EPUB 4 is defined as a profile (-> a subclass, a specialization) of a PWP, and if a PWP imposes a landing page, therefore it will be logically impossible to make the landing page optional in the EPUB.

That's why I consider that the landing page in a PWP is to be optional. And this is not a big deal, as  it can be created automatically, even on the fly, for the re-hydreated WP <-> your comment "a landing page may be an empty page with a link to the json manifest".

At this moment, we are talking about a WP, though.

**Comment by TzviyaSiegman** [comment]
> If the publication only exists as a PWP/EPUB4

Let's be careful not to conflate EPUB 4 with WP. We are talking about WP in this thread.

**Comment by llemeurfr** [comment]
> @llemeurfr, I must admit that is not the way I view things. For me, the landing page is an integral part of a WP, and is independent of the page set up by, in this case, a distributor. It may contain information that the publisher want to convey, but it is a bit like the cover page of a usual book.

I didn't state anything against this. A landing page is an integral part of a WP? ok. It is a bit like the cover page of a usual book? yes: it is what is "communicated" to a user as entry point into the publication.

I changed my mind about something else: I prefer considering that a landing page is not _necessarily_ in the s**** . After all if a publisher wants to make the original TOC of a book the landing page, he should be able to do that.

(I don't see why you don't want to hear about the terme "spine" anymore, "reading order" is not a collection_; we can say pages are **in** reading order, but not pages are **part of the** reading order, do we?).

**Comment by iherman** [comment]
@llemeurfr, we never defined what we mean by profile. Indeed, *in general*, it is usually a bit like a subclass, but we have the liberty to define as we want, even changing the term we use, if needed.

**Comment by iherman** [comment]
> I changed my mind about something else: I prefer considering that a landing page is not necessarily in the s**** . After all if a publisher wants to make the original TOC of a book the landing page, he should be able to do that.

I don't think we ever said that. Only that it is one of the resources that make up a WP.

**Comment by GarthConboy** [comment]
So it sounds like we have semi-agreement on some sort of HTML "landing page" that has metadata that points to the JSON manifest (this latter part is what makes the "its' a WP" identification).

We're not agreed on whether the "landing page" is part of the WP and what it must or may contain.

I was originally going to suggest we give it (the landing page) a mission... we've previously agreed that a WP must have a default reading order (what us EPUB-ers would call a spine), so why not MUST this into the landing page?

But, on further reflection this doesn't really work well:

-- The manifest then wouldn't fully define the publication unless it back-pointed to the landing page
-- The default reading order would then need to be renderable, and perhaps pretty, for a reasonable landing page
-- `HTML/<nav>` would need to be machine processable for a Reading System to make sense to the default reading order

So, I'll not make that suggestion.   And suggest instead:

-- The landing page points to the JSON manifest (and thus makes the "its' a WP" identification)
-- The landing page can contain arbitrary content, hopefully somehow "about" the book
-- The landing page may, or may not, be part of the publication (as determined by the content of the JSON manifest)
-- The JSON manifest would contain the full information about the publication (the default reading order, the metadata, and hopefully secondary resource info, et al)

**Comment by llemeurfr** [comment]
@GarthConboy this make total sense to me.

In the third assertion (The landing page may, or may not, be part of the publication (as determined by the content of the JSON manifest)) I would replace "publication" (not defined in the spec) by resource list (https://w3c.github.io/wpub/#wp-resource-list).

**Comment by iherman** [comment]
@GarthConboy,

My only reservation, as you might expect, is whether the landing page is part of the corresponding WPUB or not; I agree with your other points. The landing page is the page where I am led to if I dereference the WPUB's address and/or, in many cases, the WPUB's canonical identifier (e.g., its DOI if I have a scholarly paper), and it looks unnatural for me to get not to the publication itself (like the cover page) but to another page that is not even offlined with the publication (provided, which I think is a safe supposition, the what we are offlining is what appears in the publication's resource list). I would at least prefer a "The landing page SHOULD be part of the publication's resource list" (which allows for WPUB's that do not do that in the case there are really good arguments against it).

That being said, I am ready to go with the majority.

**Comment by HadrienGardeur** [comment]
I come to a fairly similar conclusion to what @GarthConboy said but from a different angle.

While @prototypo said that the landing page must identify itself as a Web Publication, I believe that a link back to a manifest has a different meaning/purpose.
IMO, the only way to know that a given resource belongs to a publication is to fetch the manifest for that publication and extract its reading order and list of ancillary resources from it:

- we can't rely on the presence of a link instead, not all resources of a publication will contain that link
- we also can't have a secondary way of establishing that a resource belongs to a publication, otherwise we could have situations where a resource contains a link to a manifest but is not listed in in any way in the manifest or the TOC

Therefore the role of the link to the manifest is not to establish that the current resource belongs to a given Web Publication, but instead to provide auto-discovery (similar to how discovery works in RSS/Atom).

Discovery shouldn't be limited to resources contained in a publication, it should be possible to discover a WP any place where it makes sense to do so.

The second question that we need to ask ourselves about this "landing page" (still unhappy with this concept and term), is whether we'll always have the ability to access a manifest from it.

I'm not entirely sure that's the case.

Let's say that I want to access the latest issue of _The New Yorker_, where each issue is now published as a Web Publication.
These issues are behind a paywall, I either need to pay for the individual issue or log in to prove that I'm a subscriber.

We can assume that most resources from the Web Publication will remain inaccessible (behind a redirect and/or a 401 HTTP status code) as long as I haven't gone through a number of steps. But what about the landing page itself? What about the manifest?

The "landing page" seems like a good place to handle such a transaction (buy the issue or log in) but I'm not sure that it's in the best interest of a user agent and/or a publisher to expose a Web Publication manifest to a user that won't be able to access its resources.

I think that's a very common scenario and many publications will be behind such a paywall, no matter what the nature of the transaction is (buying a book from a publisher, subscription to a magazine, borrowing a publication from a library).

**Comment by mattgarrish** [comment]
If the publication is locked up and the publisher wants the user to go to a marketing page if they don't have access, wouldn't they just put a redirect on the landing page?

Wouldn't you use an acquire link for what you're describing Hadrien, not the address?

After I buy the publication, I don't want to end up back at that marketing page again if I follow the entry link. I don't see the address precluding there being other points of entry; it's just one solid link back to the publication, and from which the publication can be initiated. Not having it be a resource of the publication also strikes me as a strange, but I would agree it doesn't necessarily have to be in the default reading order, since that's optional to begin with.

**Comment by HadrienGardeur** [comment]
That's one way to deal with this issue @mattgarrish, but it also means that a user agent can't reliably use a WPub URL to start reading a publication or "discover" it in the way that's described in this issue.

If user A (subscriber) shares the URL of the WPub with user B (not a subscriber), there could be multiple steps before user B can start reading the publication at all.

Regarding discovery: it's the same concept for RSS/Atom or a PWA with the Web Application Manifest.
It's possible to discover an app through its marketing page, where the Web Application Manifest will point to a completely different resource as the `start` URL.

**Comment by mattgarrish** [comment]
No, not if they don't have access, but can we really solve that only with one a link?

To me, the address firmly plants you in the publication and that's what makes it unique. Pages that help you acquire and access that page are necessary scaffolding, but they aren't the publication.

Would it help to say in the section that the landing page is not required to be an unprotected page and that authors need to ensure that users who don't have access to read the publication are taken to a more appropriate location and/or to consider the inclusion of technologies like acquire links depending on the context in which the publication is distributed?

**Comment by TzviyaSiegman** [comment]
I think we have gone a bit off topic. The sole question we are trying to answer is what happens when I dereference a WPUB URL. To my mind, this MUST (or at least SHOULD) be a part of the publication. See Ivan's [comment](https://github.com/w3c/wpub/issues/94#issuecomment-344221304) as well. Otherwise, why is this the URL of the publication?

**Comment by mattgarrish** [comment]
I agree, but I wonder if "landing page" invites confusion. A product landing page isn't necessarily the product. It might well only be advertising on how to acquire it.

The only alternative I can think of would "(publication) entry page" as it maybe has a little less marketing connotations to it. Entry pages don't have to be the home page, or in our case the first resource in the reading order.

**Comment by iherman** [comment]
Following up on @mattgarrish's [comment](https://github.com/w3c/wpub/issues/94#issuecomment-344272182), taking a very specific journal publication as an example.

The https://www.lemonde.fr is the URL (identifier?) of the newspaper 'Le Monde'. It could be considered as a Web Publication; it gives the cover page of the newspaper, and would also link to a manifest listing all the _publicly accessible_ articles.

The https://abonnes.lemonde.fr/ is the URL of the same newspaper but for subscribers. It could be considered as a _different_ Web Publication, giving access to all the articles.

The two WP-s share a number of resources. If I arrive to a browser typing in https://abonnes.lemonde.fr/ (and I am not yet logged in) then there is a redirect to https://secure.lemonde.fr/sfuser/connexion which asks me to log in. If I do, I get to the the WP itself.

I do not see anything in this scheme that would contradict referring to https://abonnes.lemonde.fr/ as _the_ address for that specific WP, and for which the lemonde.fr server will return (eventually, possibly after the login process)  that landing page (for a lack of a better word for now).

---

The discovery aspect is different. We actually did discuss, and proposed a decision on, the relevant issue at TPAC; it cannot be _REQUIRED_ for each resource in a WP to include a link to its manifest (or links to its manifests, as the example above would require). It would be a management nightmare in some cases to ask that, if at all possible (the "editor" of a WP may not have write access to each of the resources). The current term is to use _SHOULD_ for that purpose. The only reliable way to discover is the URL (address, maybe canonical ID) of the publication.

**Comment by dauwhe** [comment]
Agree with @mattgarrish that "landing page" is not a good term. 

I've been reading issues in the web app manifest spec, which is quite confusing. There are several applicable concepts in that spec. 

1. The URL of the  HTML page which contains a link to the manifest, which is called the "Document URL".
2. The URL of the manifest itself, the "Manifest URL"
3. The URL of the start page/entry point to the web app, which is the "Start URL" (we've been calling this the "landing page" although we probably shouldn't).
4. The scope of the web app, the "Scope URL"

Rules:

1. Scope URL must be same origin as Document URL
2. Start URL must be within the scope of the Scope URL
3. Start URL must be same origin as Document URL

So the following would be possible, I think:

document URL: `example.com/Foo/index.html` (contains link to `example.com/MobyDick/manifest.json`)

manifest URL: `example.com/MobyDick/manifest.json` (with scope `/MobyDick/`)

start URL: `example.com/MobyDick/title-page.html`

So if `title-page.html` did not have a link to the manifest, you would never know it was part of a web app. The only way to get app behavior would be to navigate to `example.com/Foo/index.html`.

But this feels more complicated than necessary, and quite unintuitive. Life would be much simpler if the start URL was the same as the document URL.

**Comment by BigBlueHat** [comment]
@dauwhe if the URL of the Web Publication is `example.com/Foo/` then it can return whatever default resource it wants there. That resource MUST identify itself as a Web Publication (if it is one).

It MUST also provide *some means* of knowing the locations of any other primary resources (if any). Those other primary resources, if accessed directly, MAY point back to the Web Publication they're apart of (but they might not).

When my User Agent GET's the Web Publication, it needs an "entry point" (tnx @mattgarrish) and if that's in HTML, we can shim all the things. If it's in anything else, then we'll be waiting eternally for someone to implement it...

**Comment by HadrienGardeur** [comment]
> I think we have gone a bit off topic. The sole question we are trying to answer is what happens when I dereference a WPUB URL.

I think that this is very much a core part of what we need to discuss, access to a given resource and real world examples are two things that are rarely (if ever) mentioned in this group.

A large number of publications will be behind a paywall, if what we're designing now doesn't work for those publications, we're heading straight towards something that could be unusable.

> I do not see anything in this scheme that would contradict referring to https://abonnes.lemonde.fr/ as the address for that specific WP, and for which the lemonde.fr server will return (eventually, possibly after the login process) that landing page (for a lack of a better word for now).

@iherman, but this means that following the WPub URL won't _directly_ open the publication itself and/or provide a link to the manifest.

> I've been reading issues in the web app manifest spec, which is quite confusing. There are several applicable concepts in that spec.

@dauwhe, I provided a similar list in a previous comment of this issue:

- manifest
- start
- table of contents

... but what we're discussing for the WPub URL seems to be none of the above. It MAY be the _start URL_ or the _TOC URL_ but that's not a requirement.

**Comment by mattgarrish** [comment]
> A large number of publications will be behind a paywall, if what we're designing now doesn't work for those publications, we're heading straight towards something that could be unusable.

But is it all that big a problem? If publishers set the primary entry point, which is what this allows, it enables them to put in place the necessary prompts for obtaining the content for those who don't have access.

If I try to directly access a link in a paywalled site now, like news articles, I typically get an overlay div that sells me the content or asks me to log in. Once I do that, the site knows the url I was trying to access and sends me back to it. Do we really want to assume that there is a separate product page for every piece of content? These are problems that the publisher has to solve if they want to lock up their content, and that there are probably better solutions for than the address.

**Comment by TzviyaSiegman** [comment]
> These are problems that the publisher has to solve if they want to lock up their content, and that there are probably better solutions for than the address.

I completely agree with @mattgarrish. Further, we are not writing a spec for purchasing publications. As long as we don't hinder the sales and distribution process, which we are not (perhaps even helping it), I think we are fin.

**Comment by HadrienGardeur** [comment]
> But is it all that big a problem? If publishers set the primary entry point, which is what this allows, it enables them to put in place the necessary prompts for obtaining the content for those who don't have access.

If we require to return a page that contains:

- content from the publication (first resource in reading order and/or TOC)
- and a link to the manifest

... then sure, this could be a problem. My point is that this won't always be the case, and as you've pointed out that there will be a certain number of indirect steps before we actually get to the content itself.

**Comment by mattgarrish** [comment]
But we don't require anything to be returned. All we say is that at that at the address URL is an html page of some kind with a link element pointing to the manifest. There's no guarantee it's world readable.

I agree that we shouldn't try to over-prescribe what content has to be at that location. Advice about having navigation aids is still sound, as I could be following a link in to the publication on a user agent that doesn't support web publications, but haven't we moved away from saying any certain content should/must be present?

**Comment by HadrienGardeur** [comment]
> But we don't require anything to be returned. All we say is that at that at the address URL is an html page of some kind with a link element pointing to the manifest. There's no guarantee it's world readable.

Are we sure that the manifest itself won't be behind a redirect or 4xx code of its own?
Even if it isn't, do we want to link to a manifest if all of its resources return a 401?

**Comment by BigBlueHat** [comment]
We don't need to re-specify all the various HTTP status code use cases. What we MUST (hehe) specify is what comes back when the URL for a Web Publication is returned.

If we want this to work in the browsers of today--which don't know what a Web Publication is or what it should do with it--then the URL for a Web Publication MUST be an HTML document.

If we can agree to that much, then we can file issues for all the other things that have been mentioned in here--but let's do that *after* we make some progress.

So...
 - A Web Publication MUST be identifiable by a URL unique to the Web Publication (not one that *also* points to a constituent part)
 - A request of that URL MUST return an HTML document (because if we don't do that, we're not building within the extensible Web, but for non-Web things like ebook readers only)
 - ...and here's where we all start to disagree and where more things need to be determined. 😸

**Comment by llemeurfr** [comment]
I fear we are going into circles here, and I don't think will come to something useful as they are multiple ways to create a paywall (most are client side ways to hide content, but some are based on server side redirects, or so it seems). 

How about: 
A WP has a Start page (ok for start), with a link to the JSON manifest. It is listed in WP resources, but may or may not be in "reading order". There may be some authorization needed (paywall, authentication) to see its (full) content. If such authorization has not been acquired, it would be logical that the access to the WP manifest is also un-autorized (403); if it is authorized for any reason, access to the resources of the publication should be un-authorized (if not, why building a protected access?). This is up to the web server to enforce such protection.

**Comment by iherman** [comment]
@HadrienGardeur 

>> I do not see anything in this scheme that would contradict referring to https://abonnes.lemonde.fr/ as the address for that specific WP, and for which the lemonde.fr server will return (eventually, possibly after the login process) that landing page (for a lack of a better word for now).
>
> @iherman, but this means that following the WPub URL won't directly open the publication itself and/or provide a link to the manifest.

No, it doesn't mean that. The URL is the https://abonnes.lemonde.fr/ and, after some turnaround the server will reply on that request withthe content of that WP's landing/starting/whatever page. 

(B.t.w., the example shows that it is perfectly possible to set up a paywall around a WP. The fact that, maybe, some of _today's_ setup does not function that way is not relevant.)

>If we require to return a page that contains:
>
> * content from the publication (first resource in reading order and/or TOC)
> * and a link to the manifest
>
> ... then sure, this could be a problem.

First of all, we agreed that it is not _required_ to put content from publication. Only the link to the manifest. And I am not convinced it _is_ a problem. (I am _not_ talking about a never-on-the-Web-PWP, we have discussed that before and it is for a later discussion.)

Hadrien, let me turn things around (yes, I am sneaky). What is your answer to the question of the original issue, and the [problem statement](https://github.com/w3c/wpub/issues/94#issue-272078084) of @prototypo?

**Comment by iherman** [comment]
@llemeurfr, I guess

> if it is authorized for any reason, access to the resources of the publication should be un-authorized

should say

> if it is _unauthorized_ for any reason, access to the resources of the publication should be un-authorized

With that change I agree with what you say but, as @BigBlueHat said: 

> We don't need to re-specify all the various HTTP status code use cases. What we MUST (hehe) specify is what comes back when the URL for a Web Publication is returned.

**Comment by mattgarrish** [comment]
> With that change I agree with what you say but, as @BigBlueHat said:
> 
>> We don't need to re-specify all the various HTTP status code use cases. What we MUST (hehe) specify is what comes back when the URL for a Web Publication is returned.

Yes, this is the same rabbit hole we went down looking at the lifecycle. The more we try to solve specific scenarios the more will pop up, or more exceptions we'll think of.

**Comment by TzviyaSiegman** [comment]
I think we need to remember what we are documenting here. We are not writing a spec for a bookstore. We are writing a spec for Web Publications. As long as we do not disrupt the supply chain, I don't think discussions of pay walls etc are relevant, and I think we can close this issue with @mattgarrish https://github.com/w3c/wpub/pull/101.

**Comment by mattgarrish** [comment]
I'm about to push a change that swaps landing page for entry page, but thoughts welcome on that prose.

**Comment by BigBlueHat** [comment]
👍 to pushing ahead with #101 and making that the foundation going forward.

Once #101 is merged, I'd consider this particular issue closed and new ones (about All The Things) can be filed separately.

Great work as ever @mattgarrish!

**Comment by HadrienGardeur** [comment]
@TzviyaSiegman that's not what we've been discussing, these questions are all relevant to what can and cannot be present in such a page.

> Hadrien, let me turn things around (yes, I am sneaky). What is your answer to the question of the original issue, and the problem statement of @prototypo?

At this point, I feel that the only thing we can say is that the WPUB URL points to an HTML document.

This document:

- MAY be within the boundaries of the publication
- MAY also be in the reading order
- MAY as well contain a link to the manifest

But these are all MAY, I don't think any of these items can be a MUST.

**Comment by BigBlueHat** [comment]
@HadrienGardeur let's file those as separate issues and discuss them there. Knowing what we get back from the URL that identifies a Web Publication is a necessary foundational step to any of those.

**Comment by HadrienGardeur** [comment]
@BigBlueHat identification and discovery are two different concepts, mashing them together is tricky and can lead to various problems.

IMO the only thing that can identify that a given resource is within a WP is the manifest itself.

**Comment by BigBlueHat** [comment]
> IMO the only thing that can identify that a given resource is within a WP is the manifest itself.

@HadrienGardeur that's not what's being discussed here. We're discussing what one gets back when requesting the identifying URL of the Web Publication. That has to be HTML if we're going to teach browsers new tricks.

**Comment by HadrienGardeur** [comment]
@BigBlueHat and this is the one thing that we agree on.

**Comment by BigBlueHat** [comment]
@HadrienGardeur excellent! :grin: That's all this issue was meant to be about afaict. Everything else deserves it's own discussion space.

Let's ship #101 and go file those other issues. 😄

**Comment by HadrienGardeur** [comment]
@BigBlueHat #101 goes beyond that and I've made a comment stating exactly why.

**Comment by mattgarrish** [comment]
Let's keep sight of the fact that these are not final decisions for a specification that is about to ship. Accepting this only puts it on the radar for comments in the FPWD, which is what we need to get out shortly. That's why I didn't use the "c" word and only cited numbers. If this is the wrong decision, I have faith in the crowd.

If you open new issue(s) stating why you believe these requirements are wrong, I'll also add a pointer to the section, Hadrien.

**Comment by GarthConboy** [comment]
FWIW, I would amend @HadrienGardeur list above

> MAY be within the boundaries of the publication
> MAY also be in the reading order
> MAY as well contain a link to the manifest

with only a three character change... the final MAY should be a MUST, as we must be able to find the Publication (resource list, et al) from said Publication's landing page.

**Comment by mattgarrish** [comment]
It strikes me as wrong at some fundamental level that a page that doesn't belong to any publication can initiate any publication, which is what anything less than a must seems to suggest. It just feels like there's a trust problem inherent in such a scenario. Am I getting spoofed? Will that page appear in the publication until I  navigate somewhere else?

**Comment by HadrienGardeur** [comment]
How is that any different from the way RSS/Atom or PWA work?

I could look at a presentation page for a Web App that won't be in the Web App itself that will trigger the Chrome install banner for example.
There are other requirements (@dauwhe has listed some of them, but each browser has different expectations), but this is not fundamentally any different.

**Comment by mattgarrish** [comment]
Installing an app isn't the same thing as instantiating the publication, though. There's a scope on the web app, so any resource outside that scope couldn't pretend to belong to the app when the app is launched.

But what I find confusing in where I see this going is that sometimes a web page that links to a manifest is part of the publication and when you initiate the publication you initiate at that page. But then some other times you have a link to initiate the publication but the page you're on is not part of the publication. And you seemingly can't rely on the resource list to check this, since it has no requirement for completeness of resources or even top-level resources.

So how does a user agent know when to initiate the current page, or when it's an indirection to something else? That's the hole that this opens in my mind, perhaps more than the security, and that hole exists to some extent independent of this issue.

But once you have that hole, and because a page on one domain can instantiate a web publication on another, when does the user agent doubt the legitimacy of initiating the page?

**Comment by mattgarrish** [comment]
We need to stop overusing this issue, so not cutting off discussion by closing, but please start a new one so we can focus the discussion.
